proto/build_event_stream.proto

```diff
@@ -53,6 +53,18 @@ message BuildEventId {
     // assigned; the only meaningful operation on this field is test for
     // equality.
     int32 opaque_count = 1;
+
+    // BUILDBUDDY-SPECIFIC FIELDS BELOW.
+
+    // The invocation step ID. This can be used to group progress streams into
+    // discrete "steps". At most one step's stream may be open at once - the
+    // server may finalize a stream when transitioning from one stream ID to
+    // another, and may return an error if attempting to resume a previously
+    // finalized stream.
+    //
+    // BuildBuddy will use this ID to render each step's output for a workflow
+    // invocation.
+    int32 step_id = 1000;
```

Comment:
Is there a reason why this is an id while the other messages use step_number? It might be easier to string messages together if they are consistent